internal/kibana/deployment.go

```diff
@@ -11,9 +11,13 @@ import (
 // NewDeployment stubs an instance of a Deployment
 func NewDeployment(deploymentName string, namespace string, loggingComponent string, component string, replicas int32, podSpec core.PodSpec) *apps.Deployment {
 	labels := map[string]string{
-		"provider":      "openshift",
-		"component":     "kibana",
-		"logging-infra": "kibana",
+		"provider":                     "openshift",
```

Comment:
The labels variable is used for both the labels of the Deployment and Pod resources as well as in the LabelSelector in the Deployment.

Selectors are write-once attributes, so this will create an error when the CLO tries to update an existing Deployment, which would make updates from a previous version impossible.

I think you need to split up the labels and the LabelSelector, so that the selector can stay with the current value to avoid this error. Alternatively you need to make sure that the old resource is deleted before the update (I would not go that route, though).

Reply:
@xperimental  Can you suggest what to change in order to get a label on elasticsearch pod

Reply:
> I think you need to split up the labels and the LabelSelector, so that the selector can stay with the current value to avoid this error. Alternatively you need to make sure that the old resource is deleted before the update (I would not go that route, though).

Keep the set of labels used for the selector as it currently is and only change the labelset used for the object. Not all labels of the pod have to be in the selector. Usually it is a good idea to limit the selector to the smallest set of labels needed to uniquely identify the pods. But once a set for the selector is chosen there's no easy way to change it.